src/core/killall.c

```diff
@@ -46,11 +50,9 @@ static bool ignore_proc(pid_t pid, bool warn_rootfs) {
         if (!f)
                 return true; /* not really, but has the desired effect */
 
-        count = fread(&c, 1, 1, f);
-
-        /* Kernel threads have an empty cmdline */
-        if (count <= 0)
-                return true;
+        /* Try to read the first character of the command line. If the cmdline is empty (which might be the case for
+         * kernel threads but potentially also other stuff), we don't care much. */
+        (void) fread(&c, 1, 1, f);
```

Comment:
This leaves `c' undefined if command is empty.

Reply:
yes, that's why I added the comment and initialize c explicitly to zero first, so that it doesn't matter, and we aren't relying in undefined behaviour.